### PR TITLE
SPM-1049: Advisories sync mismatch metric

### PR DESCRIPTION
**Pull request by AlesKas**
## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [x] Input Validation
- [x] Output Encoding
- [x] Authentication and Password Management
- [x] Session Management
- [x] Access Control
- [x] Cryptographic Practices
- [x] Error Handling and Logging
- [x] Data Protection
- [x] Communication Security
- [x] System Configuration
- [x] Database Security
- [x] File Management
- [x] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: vmaas_sync/advisory_sync.go [hunk 1]
```diff
@@ -286,6 +287,8 @@ func checkAdvisoriesCount() error {
 
 	errataCount := int64(errataResponse.GetPages()) + 1
 	if databaseAdvisoriesCount != errataCount {
+		mismatch := errataCount - databaseAdvisoriesCount
+		advisoriesCountMismatch.Add(math.Abs(float64(mismatch)))
```
Comment (Josca) [inline vmaas_sync/advisory_sync.go:291]: This practically shouldn't happen. I would add just error log here:
~~~go
utils.Log("mismatch", mismatch).Error("Incremental advisories sync mismatch found!").
~~~
What do you mean?

Comment (AlesKas) [reply vmaas_sync/advisory_sync.go:291]: Jiří thought it would be nice to have and see it in Grafana, we can discuss this.

Comment (Josca) [reply vmaas_sync/advisory_sync.go:291]: @AlesKas ok, so can you at least add this log line, please?

**Review by Josca — APPROVED** [review]
/lgtm

**Pull request merged**